extra_model/_run.py

```diff
@@ -21,24 +41,32 @@ def run(
     """Docstring."""
     logging.basicConfig(format="  %(message)s")
 
-    extra_model = ExtraModel(models_folder=embeddings_path)
-    extra_model.load_from_files()
-
     logger.info(f"Loading data from {input_path}")
     input_data = pd.read_csv(input_path)
-    if not {"CommentId", "Comments"}.issubset(input_data.columns):
-        raise ExtraModelError(
-            f"Input columns must include `CommentId` and `Comments`, \
-        but got {input_data.columns.to_list()} instead"
-        )
 
-    logger.info("Running `extra-model`")
-    results_raw = extra_model.predict(comments=input_data.to_dict("records"))
-    results = pd.DataFrame(results_raw)
+    check_input_structure(input_data)
+    results = build_model_and_predict(embeddings_path, input_data)
```

Comment:
then this will be only `results = run_from_dataframe(input_data, embeddings_path)`